server/src/main/java/org/apache/iotdb/db/qp/executor/AbstractQueryProcessExecutor.java

```diff
@@ -91,11 +104,167 @@ private QueryDataSet processShowQuery(ShowPlan showPlan) throws QueryProcessExce
         return processShowFlushTaskInfo();
       case VERSION:
         return processShowVersion();
+      case TIMESERIES:
+        return processShowTimeseries((ShowTimeSeriesPlan) showPlan);
+      case STORAGE_GROUP:
+        return processShowStorageGroup();
+      case DEVICES:
+        return processShowDevices();
+      case CHILD_PATH:
+        return processShowChildPaths((ShowChildPathsPlan) showPlan);
+      case COUNT_TIMESERIES:
+        return processCountTimeSeries((CountPlan) showPlan);
+      case COUNT_NODE_TIMESERIES:
+        return processCountNodeTimeSeries((CountPlan) showPlan);
+      case COUNT_NODES:
+        return processCountNodes((CountPlan) showPlan);
       default:
         throw new QueryProcessException(String.format("Unrecognized show plan %s", showPlan));
     }
   }
 
+  private QueryDataSet processCountNodes(CountPlan countPlan) throws SQLException {
+    List<String> nodes = getNodesList(countPlan.getPath().toString(), countPlan.getLevel());
+    int num = nodes.size();
+    List<Path> paths = new ArrayList<>();
+    paths.add(new Path(COLUMN_COUNT));
+    List<TSDataType> dataTypes = new ArrayList<>();
+    dataTypes.add(TSDataType.INT32);
```

Comment:
It would be better to use `Collection.singletonList` and `Arrays.asList` to create lists that all elements are known as the lists will be more compact.

Reply:
fixed.